### PR TITLE
Viability phase

### LINKED ISSUE #61: Viability Phase
Should use data from #60 to choose what _Learning Profiles_ (AL / ML / Dataset)  are best.

### PR DESCRIPTION
**Pull request by Ludvig-Joborn**
Closes #61

### CODE REVIEW

File: src/api/storage.py [hunk 1]
```diff
@@ -104,8 +122,19 @@ def get_labels(self):
         Returns:
             np.ndarray: A 2D numpy array with labels for arousal and valence.
```
Comment (MattiasBeming) [inline src/api/storage.py:123]: Update docstring

File: src/learning_profile.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
+from api import storage
+import numpy as np
+import hashlib
+
+
+class LearningProfile:
+
+    def __init__(self, train_dataset_name: str, test_dataset_name: str,
```
Comment (MattiasBeming) [inline src/learning_profile.py:8]: typing?

File: src/viability_phase.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+    # Get song ids of full training data
+    full_train_song_ids = full_train_ds.get_contained_song_ids()
+
+    ##########
+    # Step 1 #
+    ##########
+
+    # Initialize training songs with first song id
+    seed_song_ids = [full_train_ds.get_data().iloc[0].index.values[0]]
```
Comment (MattiasBeming) [inline src/viability_phase.py:75]: use first song id in full_train_song_ids ?

File: src/viability_phase.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    unlabeled_train_song_ids = list(
+        set(full_train_song_ids).difference(seed_song_ids))
+
+    # Get dataframe and labels of full training data
+    full_train_dat = full_train_ds.get_data()
+    full_train_ar = full_train_ds.get_arousal_labels()
+    full_train_va = full_train_ds.get_valence_labels()
+
+    # Derive number of samples per song (assume same for all songs)
+    n_samples_per_song = full_train_dat.loc[seed_song_ids, :].shape[0]
```
Comment (MattiasBeming) [inline src/viability_phase.py:87]: seed_song_ids[0] -> to make sure one song is selected

**Review by MattiasBeming — CHANGES_REQUESTED** [review]

File: src/api/storage.py (hunk 1 above)
Comment (Ludvig-Joborn) [reply src/api/storage.py]: Fixed!

File: src/learning_profile.py (hunk 2 above)
Comment (Ludvig-Joborn) [reply src/learning_profile.py:8]: Resolve for now.

File: src/viability_phase.py (hunk 3 above)
Comment (Ludvig-Joborn) [reply src/viability_phase.py:75]: Works anyway.

File: src/viability_phase.py (hunk 4 above)
Comment (Ludvig-Joborn) [reply src/viability_phase.py:87]: Not necessary.

**Review by MattiasBeming — APPROVED** [review]
Nice!

**Pull request merged**